chroma_core/models/stratagem.py

```diff
@@ -0,0 +1,399 @@
+# Copyright (c) 2019 DDN. All rights reserved.
+# Use of this source code is governed by a MIT-style
+# license that can be found in the LICENSE file.
+
+import logging
+import json
+from toolz.functoolz import pipe, partial, flip
+
+from django.db import models
+
+from chroma_core.lib.cache import ObjectCache
+from chroma_core.models.jobs import StatefulObject
+from chroma_core.lib.job import Step, job_log, DependOn, DependAll, DependAny
+from chroma_core.lib.stratagem import (
+    parse_stratagem_results_to_influx,
+    record_stratagem_point,
+    clear_scan_results,
+    temp_stratagem_measurement,
+    stratagem_measurement,
+    aggregate_points,
+    submit_aggregated_data,
+)
+
+from chroma_core.models import Job
+from chroma_core.models import StateChangeJob, StateLock, StepResult, LustreClientMount
+from chroma_help.help import help_text
+from chroma_core.models import (
+    AlertStateBase,
+    AlertEvent,
+    ManagedHost,
+    ManagedMdt,
+    ManagedTarget,
+    ManagedTargetMount,
+    Volume,
+    VolumeNode,
+    StorageResourceRecord,
+)
+
+
+class StratagemConfiguration(StatefulObject):
+    filesystem_id = models.IntegerField(
+        help_text="The filesystem id associated with the stratagem configuration", null=False
+    )
+    interval = models.IntegerField(help_text="Interval value in seconds between each stratagem execution", null=False)
+    report_duration = models.IntegerField(
```

Comment:
Can't these be null? If so, do we need the active flags?

Reply:
yes, updated